### PR TITLE
move exporting keystore json to background queue #5646

### LINKED ISSUE #5646: Exporting Keystore JSON blocks UI for a long time
`LegacyFileBasedKeystore.convertPrivateKeyToKeystoreFile` contains an operation that takes a while. Maybe modify it to return a Promise and for the long operation in `convertPrivateKeyToKeystoreFile()` to be in a separate queue? Careful with threading. Maybe just this one is slow?

```
try KeystoreKey(password: passphrase, key: privateKey)
```

Is it safe to move away from main?

### PR DESCRIPTION
**Pull request by oa-s**
Closes #5646

### CODE REVIEW

**Review by hboon — COMMENTED** [review]
Changes look a bit more extensive and affects key management. I'll look at this later.

**Review by hboon — APPROVED** [review]
Looks great. Thanks!

File: AlphaWallet/Backup/Coordinators/BackupCoordinator.swift [hunk 1]
```diff
@@ -40,14 +42,14 @@ class BackupCoordinator: Coordinator {
...
-        )
+        navigationController.displayLoading(text: R.string.localizable.exportPresentBackupOptionsLabelTitle())
+
         let prompt = R.string.localizable.keystoreAccessKeyHdVerify()
-        keystore.exportRawPrivateKeyForNonHdWalletForBackup(forAccount: account, prompt: prompt, newPassword: newPassword) { [weak self] result in
-            guard let strongSelf = self else { return }
-            strongSelf.handleExport(result: result, completion: completion)
-        }
+        keystore.exportRawPrivateKeyForNonHdWalletForBackup(forAccount: account, prompt: prompt, newPassword: newPassword)
+            .sink { [weak self] result in
```
[the remark this answers is not in the document]
Comment (oa-s) [reply AlphaWallet/Backup/Coordinators/BackupCoordinator.swift:49]: we already return result of `exportRawPrivateKeyForNonHdWalletForBackup` in `.main`

File: modules/AlphaWalletFoundation/AlphaWalletFoundation/KeyManagement/EtherKeystore.swift [hunk 2]
```diff
@@ -310,80 +310,89 @@ open class EtherKeystore: NSObject, Keystore {
...
-            completion(.failure(.failedToExportSeed))
-        case .userCancelled:
-            completion(.failure(.userCancelled))
-        case .notFound, .otherFailure:
-            completion(.failure(.failedToExportSeed))
-        }
+                //Careful to not replace the if-let with a flatMap(). Because the value is a Result and it has flatMap() defined to "resolve" only when it's .success
+                let result = legacyFileBasedKeystore.export(privateKey: key, newPassword: newPassword)
+                return .just(result)
+            }.receive(on: RunLoop.main)
```
[the remark this answers is not in the document]
Comment (oa-s) [reply modules/AlphaWalletFoundation/AlphaWalletFoundation/KeyManagement/EtherKeystore.swift:333]: yep

File: AlphaWallet/Backup/Coordinators/BackupCoordinator.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (hboon) [reply AlphaWallet/Backup/Coordinators/BackupCoordinator.swift:49]: Sorry, I thought I deleted this comment after typing this: https://github.com/AlphaWallet/alpha-wallet-ios/pull/5650#discussion_r1017402343. It's better anyway.

**Pull request merged**